Src/DependencyCollector/Shared/Implementation/DependencyDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,308 @@
+namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Web.Implementation;
+
+    internal class DependencyDiagnosticSourceListener : IObserver<KeyValuePair<string, object>>, IDisposable
+    {
+        private const string DataSourceNameSuffix = ".Monitoring";
```

Comment:
Noise and performance. This is not a generic DataSource listener - this is a generic __RDD__ DS listener, so we do not want to try to convert any event/activity to a dependency as most of them will be unwanted